### DISCUSSION TITLE
problema installazione MQTT su H.A.

### DISCUSSION THREAD

**Discussion by gericom90** [indomus/forum#1249] [HUB: Home Assistant]
un saluto a tutti,  sto installando Home Assistant Core su un Raspberri Pi4, seguendo la seguente guida   
https://indomus.it/guide/come-installare-e-configurare-il-broker-mqtt-mosquitto-su-raspberry-pi/
Tutto ok fino al comando     sudo dpkg -i libwebsockets3_1.2.2-1_armhf.deb    che genera il seguente errore:
-------------
pi@raspberrypi:~ $ sudo dpkg -i libwebsockets3_1.2.2-1_armhf.deb
dpkg: problemi con le pre-dipendenze di libwebsockets3_1.2.2-1_armhf.deb contenente libwebsockets3:armhf:
 libwebsockets3 pre-dipende da multiarch-support
  multiarch-support non è installato.
dpkg: errore nell'elaborare l'archivio libwebsockets3_1.2.2-1_armhf.deb (--install):
 problema con le pre-dipendenze - libwebsockets3:armhf non viene installato
Si sono verificati degli errori nell'elaborazione:
 libwebsockets3_1.2.2-1_armhf.deb
----------------
Da una ricerca in rete non ho trovato soluzioni.
Potete darmi indicazioni al riguardo?
grazie.

**Comment by snakuzzo** [comment]
Non si capisce niente.
Per favore sistema il post usando il markdown per codice e/o output dei comandi

**Comment by gericom90** [comment]
Ok, cerco di essere più chiaro.
Ho installato H.A.Core su un Raspberry PI4, ed il tutto funziona.
Sto ora cercando di installare MQTT seguendo la guida:
https://indomus.it/guide/come-installare-e-configurare-il-broker-mqtt-mosquitto-su-raspberry-pi/
Arrivo senza problemi fino al comando
 
![immagine](https://user-images.githubusercontent.com/84856140/147877555-62873464-c994-40e8-bba4-bf06850e3d46.png)

dove ottengo un errore connesso alle pre-dipendenze  

![immagine](https://user-images.githubusercontent.com/84856140/147877719-67bfa9c4-611d-472c-9044-623daa6e399e.png)

grazie per l'aiuto.

**Reply by snakuzzo** [reply]
La guida dobbiamo rivederla.
Nel frattempo...seguendo le indicazioni della guida ufficiale ([QUI](https://mosquitto.org/blog/2013/01/mosquitto-debian-repository/)).
Ti basta:
- aggiungere la key
```
wget http://repo.mosquitto.org/debian/mosquitto-repo.gpg.key
sudo apt-key add mosquitto-repo.gpg.key
```
- aggiungere la repo (una di queste in base alla distro che hai usato) 

```
cd /etc/apt/sources.list.d/
```

e poi...
```
sudo wget http://repo.mosquitto.org/debian/mosquitto-jessie.list
sudo wget http://repo.mosquitto.org/debian/mosquitto-stretch.list
sudo wget http://repo.mosquitto.org/debian/mosquitto-buster.list 
sudo wget http://repo.mosquitto.org/debian/mosquitto-bullseye.list
```

- Poi `apt-get update`
- Infine `apt-get install mosquitto mosquitto_clients`

**Reply by gericom90** [reply]
Ok snakuzzo, grazie per le indicazioni.
Tutto Ok fino al comando    apt-get update   che evidenzia dei problemi:

![immagine](https://user-images.githubusercontent.com/84856140/147886894-62354674-8e4c-4e98-9fbe-f0297a3bda9b.png)

Se controllo il file Lock  ha dimensione pari a 0

![immagine](https://user-images.githubusercontent.com/84856140/147886942-146ae55c-98e9-4ce6-95f4-1b5285ffb750.png)

**Reply by snakuzzo** [reply]
Devi lanciare i comandi apt-get con `sudo`

Es.

```
sudo apt-get update
sudo apt-get install mosquitto mosquitto_clients
```

P.S. da queste domande vedo che non sei proprio a tuo agio. Sicuro di voler proseguire con questo tipo di installazione?

**Reply by gericom90** [reply]
Ciao e grazie per le indicazioni.
In effetti le mie conoscenze sono pari a quelle di un bradipo, ma vengo da una esperienza con un Raspberry PI3 con Hassio, ed avendo acquistato un PI4 ero curioso di vedere le differenze con una installazione "core". 
In relazione al mancato comando "sudo", avendo inserito nel registro "sshd_config" l'istruzione "PermitRootLogin yes" (solo in questa fase di installazione, e quindi da togliere a completamento) ero convinto che anteporre "sudo" all'istruzione non fosse necessario.
Ancora grazie e auguri per un buon 2022

**Reply by snakuzzo** [reply]
`PermitRootLogin` non c'entra assolutamente niente con sudo
